specs/altair/validator.md

```diff
@@ -0,0 +1,367 @@
+
+A validator should plan for future sync committee assignments by noting which sync committee periods they are selected for participation.
+Specifically, a validator should:
+* Upon (re)syncing the chain and upon sync committee period boundaries, check for assignments in the current and next sync committee periods.
+* If the validator is in the current sync committee period, they can perform the responsibilities below for sync committee rewards.
+* If the validator is in the next sync committee period, they should wait until the next `EPOCHS_PER_SYNC_COMMITTEE_PERIOD` boundary and then perform the responsibilities throughout that period.
+
+## Beacon chain responsibilities
+
+A validator maintains the responsibilities given in the Phase 0 document.
+
+Block proposals are modified to incorporate the sync committee signatures as detailed below.
+
+When assigned to a sync committee, validators have a new responsibility to sign beacon block roots during each slot of the sync committee period.
+These signatures are aggregated and routed to the proposer over gossip for inclusion into a beacon block. 
+Assignments to a particular sync committee are infrequent at normal validator counts; however, an action every slot is be required when in the currently active sync committee.
+
+### Block proposal
+
+Refer to the phase 0 document for the majority of the [block proposal responsibility](../phase0/validator.md#block-proposal).
+The validator should follow those instructions to prepare a `SignedBeaconBlock` for inclusion into the chain. All changes are additive to phase 0 and noted below.
+
+#### Preparing a `BeaconBlock`
+
+No change to [Preparing for a `BeaconBlock`](../phase0/validator.md#preparing-for-a-beaconblock).
+
+#### Constructing the `BeaconBlockBody`
+
+Each section of [Constructing the `BeaconBlockBody`](../phase0/validator.md#constructing-the-beaconblockbody) should be followed. 
+After constructing the `BeaconBlockBody` as per that section, the proposer has an additional task to include the sync committee signatures:
+
+##### Sync committee
+
+The proposer receives a number of `SyncCommitteeContribution`s (wrapped in `SignedContributionAndProof`s on the wire) from validators in the sync committee who are selected to partially aggregate signatures from the full committee (see below for details).
+
+The proposer collects these contributions for further aggregation when preparing a block. 
+Recall `block.body.sync_aggregate.sync_committee_bits` is a `Bitvector` where the `i`th bit is `True` if the corresponding validator in the sync committee has produced a valid signature, and that `block.body.sync_aggregate.sync_committee_signature` is the aggregate BLS signature combining all of the valid signatures.
+
+Given a collection of `contributions` and the `BeaconBlock` under construction, the proposer processes them as follows:
+
+```python
+def process_sync_committee_contributions(state: BeaconState, 
```

Comment:
If they use this function, you need to dictate that contributions *must* be non-overlapping

Reply:
Might be easiest to dictate that they take the *best* contribution from each subcommittee

Reply:
ah yes, when i wrote this i was thinking we would end up w/ contiguous subcommittees which we don't have with the current definition of `compute_subnet_for_sync_committee`.  i think this functions needs to be updated; if we use the aggregates in the `state` we will move to non-overlapping subsets by default so i think best to leave this function alone